src/kOS/Function/Misc.cs

```diff
@@ -27,7 +27,40 @@ public override void Execute(SharedObjects shared)
             shared.Screen.Print(textToPrint);
         }
     }
+    
+    [Function("hudtext")]
+    public class FunctionHudText : FunctionBase
+    {
+        public override void Execute (SharedObjects shared)
+
+        {
+            bool      echo      = Convert.ToBoolean(shared.Cpu.PopValue());
+            RgbaColor rgba      = GetRgba(shared.Cpu.PopValue());
+            int       size      = Convert.ToInt32 (shared.Cpu.PopValue ());    
+            int       style     = Convert.ToInt32 (shared.Cpu.PopValue ());
+            int       delay     = Convert.ToInt32 (shared.Cpu.PopValue ());   
+            string    textToHud = shared.Cpu.PopValue ().ToString ();
+            string   htmlColour = rgba.ToHTMLString();
+            {                            
+            if (style == 1) {
+                ScreenMessages.PostScreenMessage("<color=" + htmlColour + "><size=" + size + ">" + textToHud + "</size></color>",delay,ScreenMessageStyle.UPPER_LEFT);
+            } else if (style == 2) {
+                ScreenMessages.PostScreenMessage("<color=" + htmlColour + "><size=" + size + ">" + textToHud + "</size></color>",delay,ScreenMessageStyle.UPPER_CENTER);
+            } else if (style == 3) {
+                ScreenMessages.PostScreenMessage("<color=" + htmlColour + "><size=" + size + ">" + textToHud + "</size></color>",delay,ScreenMessageStyle.UPPER_RIGHT);
+            } else if (style == 4) {
+                ScreenMessages.PostScreenMessage("<color=" + htmlColour + "><size=" + size + ">" + textToHud + "</size></color>",delay,ScreenMessageStyle.LOWER_CENTER);
+            } else {
+                ScreenMessages.PostScreenMessage("*" + textToHud, 3f, ScreenMessageStyle.UPPER_CENTER);
+            }
```

Comment:
As I merge this in, I'm changing the above to a switch statement instead of an if-else ladder.

I originally started changing it to the indentation style the rest of the code uses (which I really hate, by the way.  I prefer the type you use for exactly this reason - it makes if/else ladders waste less screen real estate).

But then I realized as I was doing that that a simple switch with case 1: case 2 and s on works here too.